File: BO20426122_capstone/frontend_online_reatil_store/Frontend/online-retail-store/src/app/services/inventory.service.ts

import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Inventory } from '../models/Inventory.model';

@Injectable({
  providedIn: 'root',
})
export class InventoryService {
  private baseUrl = 'http://localhost:3333/api/inventory';

  constructor(private http: HttpClient) {}

  addInventory(inventory: Inventory): Observable<Inventory> {
    return this.http.post<Inventory>(`${this.baseUrl}`, inventory);
  }

  updateInventory(id: number, inventory: Inventory): Observable<Inventory> {
    return this.http.put<Inventory>(`${this.baseUrl}/${id}`, inventory);
  }

  getInventory(id: number): Observable<Inventory> {
    return this.http.get<Inventory>(`${this.baseUrl}/${id}`);
  }

  getInventoryByProductId(productId: number): Observable<Inventory> {
    return this.http.get<Inventory>(`${this.baseUrl}/product/${productId}`);
  }

  deleteInventory(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: BO20426122_capstone/backend_online_retail_store/Online_retail_store/Inventory/src/main/java/com/onlineretailstore/inventory/service/InventoryService.java
package com.onlineretailstore.inventory.service;

import java.util.List;

import com.onlineretailstore.inventory.entity.Inventory;

public interface InventoryService {

	public Inventory addInventory(Inventory inventory);
	public void deleteInventory(Integer inventoryId);
	public Inventory updateInventory(Integer inventoryId, Inventory inventory);
	public Inventory getInventory(Integer inventoryId);
	public Inventory getInventoryByProductId(Integer productId);
	public List<Inventory> getAllInventories();
}
